cupy/_logic/content.py

```diff
@@ -39,7 +41,57 @@ def _create_float_test_ufunc(name, doc):
     ''')
 
 
-# TODO(okuta): Implement isneginf
-
-
-# TODO(okuta): Implement isposinf
+def isneginf(x, out=None):
+    """Returns a bool array, where True if input element is negative infinity.
+    Args:
+        x (cupy.ndarray): Input array.
+    Returns:
+        cupy.ndarray: Boolean array of same shape as ``x``.
+    Examples
+    --------
+    >>> cupy.isneginf(0)
+    False
+    >>> cupy.isneginf([4, -4, numpy.inf, -numpy.inf])
+    [False, False, False, True]
+
+    .. seealso:: :func:`numpy.isneginf`
+
+    """
+
+    is_inf = cupy.isinf(x)
+    try:
+        signbit = cupy.signbit(x)
+    except TypeError as e:
+        dtype = cupy.asanyarray(x).dtype
+        raise TypeError(f'This operation is not supported for {dtype} values '
+                        'as it would be uncertain.') from e
+    else:
+        return cupy.logical_and(is_inf, signbit, out=None)
```

Comment:
```suggestion
    return cupy.logical_and(is_inf, signbit, out)
```

Reply:
Hi, @kmaehashi!
Apologies for the delay. I made all the changes as per review. But I have a doubt at this point.
Using `out` instead of `out=None` produces the following result in the local system:
```
    AssertionError: Only cupy raises error
E   
E   Traceback (most recent call last):
E     File "/home/khushi/Documents/cupy/cupy/testing/_loops.py", line 43, in _call_func
E       result = impl(*args, **kw)
E     File "/home/khushi/Documents/cupy/tests/cupy_tests/logic_tests/test_content.py", line 42, in check_unary
E       return getattr(xp, name)(a)
E     File "/home/khushi/Documents/cupy/cupy/_logic/content.py", line 97, in isposinf
E       return cupy.logical_and(is_inf, signbit, out)
E     File "cupy/_core/_kernel.pyx", line 1147, in cupy._core._kernel.ufunc.__call__
E       arg_list = _preprocess_args(dev_id, args, False)
E     File "cupy/_core/_kernel.pyx", line 130, in cupy._core._kernel._preprocess_args
E       raise TypeError('Unsupported type %s' % type(arg))
E   TypeError: Unsupported type <class 'NoneType'>
```
This is weird. Since we have already specified in the function definition, why do we need to define that again?

Am I missing something?
Thanks!

Reply:
This is a good finding! Raised an issue for this #6393.

As for now, please use `out=out` instead of `out` to workaround the issue.

Reply:
> This is a good finding! Raised an issue for this # 6393.

Thanks, @kmaehashi :)

> As for now, please use out=out instead of out to workaround the issue.

Sure, done!